### DISCUSSION TITLE
Deprecated ssh algorithms

### DISCUSSION THREAD

**Discussion by panachoi** [Anime4000/RTL960x#313] [General]
According to my scan, these firmwares (at least the one I'm running 220923) only supports some outdated/deprecated algorithms:

```
kex_algorithms(1): diffie-hellman-group1-sha1
server_host_key_algorithms(1): ssh-rsa
encryption_algorithms(1): 3des-cbc
mac_algorithms(1): hmac-sha1
compression_algorithms(1): none
```

Its okay that it supported only 1 of each type, but 3des has been deprecated for nearly a decade, and "forbidden" since 2023.

In addition the ssh version distributed with OpenWRT doesn't support 3des (perhaps others too).

Perhaps the firmware could be updated to replace 3des with aes128, if its only going to support a single encryption algorithm ?

**Comment by Anime4000** [comment]
we know the old RTL960x SDK use old Linux Kernel and pretty much old on everything.
as no SDK on hand, if anyone willing to reverse engineer and make own RTL960x MIPS tool chain in spare time, then can fix a lot old stuff.

thing is, this PON Stick only can do is bridge when configured,
some user change the default IP Address and remove host interface address to hide,
